make `init` command work with CentOS packages

TomasTomecek opened the pull request:
```
init: correctly pick up sources from CentOS

though we are not using the official get_sources.sh script (to not add
such a weird dependency to the functionality), instead we download the
archive from the internet - Source0

same drill as the previous commit, except this is for sources
```

```
init: correctly fetch spec from centos dist-git

the former code was a mess, this commit makes it clear:

 * there is a spec_dir_name class variable which says in which dir the
   spec is stored: fedora = "", centos = "SPECS"

 * the variable is now used in dist_git.absolute_spec_path
```

Fixes #1104

TomasTomecek commented:
recheck

csomh commented on packit/source_git.py:
```diff
@@ -26,11 +26,16 @@
+# TODO: we'll need to distinct b/w 8 and 9 - 9 will follow Fedora
 class CentOSDistGit(DistGit):
-    """ Fedora and CentOS dist-git differ """
+    """ CentOS dist-git layout implementation """
 
-    # we store downstream content in source-git in this subdir
-    source_git_downstream_suffix = "SPECS"
+    # spec files are stored in this dir in dist-git
+    spec_dir_name = "SPECS"
+
+    # sources are stored in this dir in dist-git
+    # this applies to CentOS Stream 8 and CentOS Linux 7 and 8
+    source_dir_name = "SOURCES"
```
Wouldn't be easier to just override `get_absolute_specfile_path()`?

TomasTomecek replied:
The main reason I implemented it this was that it felt easier to me to keep the distinction via constants with a description and have unified methods. In the end it doesn't matter to me what the exact implementation is, as long as it works.

csomh commented on tests/integration/test_source_git_generator.py:
```diff
@@ -196,3 +196,23 @@ def test_create_srcgit_requre_populated(api_instance_source_git, tmp_path: Path)
     # requre needs sphinx, so SRPM is fine
+
+
+def test_centos_cronie(api_instance_source_git, tmp_path: Path):
+    source_git_path = tmp_path.joinpath("cronie-sg")
+    # create src-git
+    source_git_path.mkdir()
+    create_new_repo(source_git_path, [])
+    sgg = SourceGitGenerator(
+        LocalProject(working_dir=source_git_path),
+        api_instance_source_git.config,
+        "https://github.com/cronie-crond/cronie",
+        upstream_ref="cronie-1.5.2",
+        centos_package="cronie",
+    )
+    sgg.create_from_upstream()
+
+    # verify it
+    subprocess.check_call(["packit", "srpm"], cwd=source_git_path)
+    srpm_path = list(source_git_path.glob("cronie-1.5.2-*.src.rpm"))[0]
+    assert srpm_path.is_file()
```
Although this will test that the solution made for the reported issue works, my main concern with such an approach is that **it's slow**. Slow tests make it more difficult to quickly itterate over code, which will make developers stop running tests while they are working on code, which will lead to prolonged time to do even the simplest changes. Whenever a slow test is created the technical dept of the project increases. This is already happening with [dist-git-to-source-git](https://github.com/packit/dist-git-to-source-git), where a similar testing approach was taken.

TomasTomecek replied:
so what are you proposing then?

csomh replied:
How about some unit tests?

jpopelka commented:
The failing sess-rec tests are not related, I filled #1109 for it.

TomasTomecek commented:
+ added a unit test
+ the integration test is skipped by default

(though I disagree with the point that the integration test case I added here is too slow - that test allowed me to implement this PR and I wanna run it in future once working more on the affected source code - the benefits of having an E2E test case are far more beneficial than 6.7s)

csomh commented:
recheck

csomh reviewed (approved):
Thank you!

TomasTomecek commented:
requre tests are passing again, nice!

@csomh thanks for the review!
